src/FluffySpoon.AspNet.LetsEncrypt/Certes/LetsEncryptRenewalService.cs

```diff
@@ -76,7 +77,34 @@ public async Task RunOnceAsync()
 			try
 			{
 				var result = await _certificateProvider.RenewCertificateIfNeeded(Certificate);
+				
+				var chain = new X509Chain
+				{
+					ChainPolicy =
+					{
+						VerificationFlags = X509VerificationFlags.AllFlags,
+						RevocationFlag = X509RevocationFlag.ExcludeRoot,
+						RevocationMode = X509RevocationMode.NoCheck
+					}
+				};
+            
+				if (chain.Build(result.Certificate))
+				{
+					_logger.LogInformation("Successfully built certificate chain");
+
+					var oldChain = Interlocked.Exchange(ref _chain, chain);
+					oldChain?.Dispose();
+				}
+				else
+				{
+					// todo: wording
+					// todo: throw?
+					_logger.LogWarning("Was not able to build certificate chain");
```

Comment:
Since the VerificationFlags is AllFlags and revocation isn't requested I'm not sure (off the top of my head) if it's possible to get a false result (the only thing that would do that is ExplicitDistrust, IIRC, and we don't currently produce that on Linux).

If VerificationFlags had its default value (no flags set => all tests tested) then this would report that the intermediate couldn't be fetched (or something was expired, or something weird was going on).  If that's an interesting state to differentiate, then just remove the assignment of VerificationFlags.

Reply:
The flags were copied from the logic that SSLStream performs

Reply:
> The flags were copied from the logic that SSLStream performs

Yeah, SslStream is just trying to build a best-effort chain and see what it can do with that.  I'm just suggesting that for logging purposes it might make sense to try to build a "normally valid" chain.